src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_IncrementalAnalyzer.cs

```diff
@@ -77,7 +77,7 @@ private async Task AnalyzeDocumentForKindAsync(TextDocument document, AnalysisKi
                 if (nonCachedStateSets.Count > 0)
                 {
                     var analysisScope = new DocumentAnalysisScope(document, span: null, nonCachedStateSets.SelectAsArray(s => s.Analyzer), kind);
-                    var executor = new DocumentAnalysisExecutor(analysisScope, compilationWithAnalyzers, _diagnosticAnalyzerRunner, logPerformanceInfo: true);
+                    var executor = new DocumentAnalysisExecutor(analysisScope, compilationWithAnalyzers, _diagnosticAnalyzerRunner, logPerformanceInfo: true, onAnalysisCanceled: OnAnalysisCanceled);
```

Comment:
❔ Can we do this without adding more per-call closure allocations?

Reply:
Sure.

Reply:
Actually, the `DocumentAnalysisExecutor` is created just once in this method, so it should already be a single allocation?